Polish mobile, notably iPhone

jasmussen opened the pull request:
This is sort of a kitchen sink PR of small fixes. It:

- Polishes the inserter tabs so the focus style isn't clippsed
- Tweaks margins in the toolbar on mobile, so things aren't shifted around
- It undoes that individually scrolling content area and makes the `body` scroll. More on this in a second.
- It polishes how the fixed-to-block toolbar looks on mobile
- It changs how the fixed position toolbars behave, making them sticky, otherwise it wouldn't work with the WP admin bar properly
- It rotates the ellipsis button that shows mobile tools so it matches the top ellipsis. This needs more work separately, with the idea of instead showing these tools when the block is selected. 
- It prevents Mobile Safari from zooming the entire page when you open the inserter (it does this because the textfield receives focus, and Mobile Safari deems the 13px font too small, so it zooms)

As such, the above fixes part of the issues described in #4077, notably the zooming and scrolling issues.


## How Has This Been Tested?

Tested in Chrome simulator, as well as the Xcode iPhone 8 simulator.

## Screenshots

![screen shot 2017-12-19 at 11 46 40](https://user-images.githubusercontent.com/1204802/34153431-56d24304-e4b2-11e7-9728-59b3d4a41fda.png)
![screen shot 2017-12-19 at 11 46 45](https://user-images.githubusercontent.com/1204802/34153435-584537fa-e4b2-11e7-88f6-151d9d78f22d.png)
![screen shot 2017-12-19 at 11 46 50](https://user-images.githubusercontent.com/1204802/34153438-5970f25e-e4b2-11e7-8c52-81d92e212a09.png)
![screen shot 2017-12-19 at 11 46 59](https://user-images.githubusercontent.com/1204802/34153442-5a71a19e-e4b2-11e7-8c14-dd56a7551228.png)

Here's a video showing the iOS simulator:

https://cloudup.com/ccvLjdy6cT3

Outside of the iOS changes, which make Gutenberg usable, I'm going to open a separate ticket about modal improvements (some work to be done there). But because these changes also affect the desktop breakpoints, we should be sure to test this PR fairly widely.

Commit:
Little polish to the top menu and inserter.

Commit:
Make mobile scroll "normally".

That is, scroll the `html` element, instead of a specially dedicated area. Although the latter has a ton of benefits, the former is the only thing that works stably on Mobile Safari. The thing is, there's overscroll bounce that you can't really turn off, so if you aren't scrolling the main content area things are just going to be all kinds of janky.

Commit:
Fix regression with "fixed to block" toolbar

Also rotate ellipsis so it matches top toolbar one.

Commit:
Prevent Safari from zooming in when opening inserter

karmatosed commented:
This is great. It does need really thorough testing though. It's important as fixes some of the experience issues 'right now' and we can then iterate.

aduth commented on editor/edit-post/layout/style.scss:
```diff
@@ -47,6 +54,11 @@
-	overflow-y: auto;
-	-webkit-overflow-scrolling: touch;
+
+	// on mobile the main content area has to scroll
```
Is this targeting iOS specifically? Does it impact larger-display iOS like iPads? Is the `-webkit-overflow-scrolling` style unnecessary if we're intending to exclude it?

jasmussen replied:
I'm essentially normalizing the scrolling behavior across the mobile breakpoint, yes. And though the iPad also suffers from the same scrolling issue, the larger screen real estate makes the scroll issue less annoying, so it'd be good to keep the property there.

youknowriad commented:
Nice work, I have some remarks:

 - You removed the margins around the block mobile toolbar, should we do the same for the "permalink toolbar" when you focus the title?

 - A small glitch, I think related to the fact that you changed how to fix the block toolbar: I expect the editor's padding/margin to stay the same if the same toolbars are shown. But there's a difference in one of the breakpoints. Take a look at the gif and see how the editor's content jumps when resizing the window.

![margin change](https://user-images.githubusercontent.com/272444/34213949-1dec2614-e5a1-11e7-9da9-96e4f4cbacb3.gif)

Commit:
Polish permalink toolbar on mobile

- Add padding when fixed toolbar is enabled so the white sheet doesn't cover the permalink bar
- Fit the permalink bar snugly to the screen edges on mobile

jasmussen commented:
Great great catches, Riad. I believe I fixed them now:

![fixes](https://user-images.githubusercontent.com/1204802/34250220-7036dfcc-e63c-11e7-8542-1bf5dee2fcf0.gif)

jasmussen commented:
Going to merge in a bit unless you have last minute objections, Riad?

Thanks.
